src/main/java/spoon/support/reflect/code/CtCaseImpl.java

```diff
@@ -166,10 +166,9 @@ private void ensureModifiableCaseExpressionsList() {
 	@Override
 	public <T extends CtStatementList> T insertBegin(CtStatementList statements) {
 		this.ensureModifiableStatementsList();
-		for (CtStatement statement : statements.getStatements()) {
-			statement.setParent(this);
-			this.addStatement(0, statement);
-		}
+		List<CtStatement> copy = new ArrayList<>(statements.getStatements());
+		statements.setStatements(null);
+		this.statements.addAll(0, copy);
```

Comment:
I'm not sure how important this is, but the `addStatement(int, CtStatement)` method does a little bit more:
a) it calls `statement.setParent(this);` for each CtStatement
b) it calls `onListAdd`

I think it is necessary to loop over all statements to add at some point to keep that behavior. I'm not sure if `List#addAll(int, Collection)` is still prefered then.
@slarse WDYT?

Reply:
You're absolutely right. I just had it in my mind that the statement list here was a model list (as it is in CtBlockImpl), but it's not. It's just an `ArrayList`.

It seems to me like the statement and expression lists in `CtCaseImpl` should be model lists as well, but I'm not sure about the design decision for them not being so currently. It could simply be that model lists were introduced at a later time, and were never added to `CtCaseImpl`. I think I'll want to wait for @monperrus or @nharrand to weigh in there before we proceed.

@Rohitesh-Kumar-Jain Can you add assertions to check that the parents of the inserted statements are set correctly? Something like so:

```java
        assertThat(firstStatementToBeInserted.getParent(), is(testCase));
        assertThat(secondStatementToBeInserted.getParent(), is(testCase));
```

Note the use of identity (`is`) as opposed to equality.

Reply:
Yes I have added these tests as well